flink-runtime/src/main/java/org/apache/flink/runtime/io/network/netty/NettyConfig.java

```diff
@@ -49,7 +49,8 @@
 
     private final InetAddress serverAddress;
 
-    private final int serverPort;
+    private final String serverPortRange;
+    private final Iterator<Integer> serverPortRangeIterator;
```

Comment:
It looks like `serverPortRangeIterator` can be created using `serverPortRange`, but is there a reason to keep both fields?

Reply:
Originally, `serverPort` was used to generate the Netty thread group name. So when I replace `serverPort` with `serverPortRangeIterator`, I also include `serverPortRange` for that purpose.

Reply:
The point of my question is, isn't one field enough? For example, if there is only the `serverPortRange` field, can't the iterator be created at the desired time using a getter method?

Reply:
Because `NetUtils.getPortRangeFromString` also does validation, the iterator is generated during configuration parsing phase, so that if there's any configuration error, the user will be notified earlier.